core/src/com/unciv/ui/components/widgets/HealthBarV0.kt

```diff
@@ -0,0 +1,317 @@
+        var colors: Array<Color> = arrayOf(Color.GREEN)
+
+        var minBarWidth: Float = 0f
+        var minBarHeight: Float = 0f
+
+        var animateDuration: Float = 0f
+        var flashingSegment: Int = -1
+        var flashDuration: Float = 1.4f
+        var flashMaxDarken: Float = 0.7f
+
+        /** If this is set, a white Label showing values and maxValue is drawn on top */
+        var fontSize = 0
+
+        /** Sets background to a uniform color and sets padding as that background includes no margins */
+        fun setBackground(color: Color, pad: Float) {
+            background = ImageGetter.getDrawable(null).tint(color)
+            this.pad = pad
+        }
+
+        /** Sets both minimum sizes (pad and background margins are added to this) */
+        fun setBarSize(width: Float, height: Float) {
+            minBarWidth = width
+            minBarHeight = height
+        }
+
+        internal fun calcMinWidth() = minBarWidth + 2 * padHorizontal +
+            (background?.run { leftWidth + rightWidth } ?: 0f)
+        internal fun calcMinHeight() = minBarHeight + 2 * padVertical +
+            (background?.run { bottomHeight + topHeight } ?: 0f)
+    }
+
+    init {
+        if (segmentCount < 2)
+            throw IllegalArgumentException("Less than two segments are nonsense")
+        if (minValue >= maxValue)
+            throw IllegalArgumentException("maxValue must be > minValue")
+    }
+
+    //region internal Fields
+    private val values = FloatArray(segmentCount - 1) { 0f }
+
+    private val drawColor = Color()
+    private val segmentColor = Color()
+    private val drawSegment = Rectangle()
+    private val whiteDot = ImageGetter.getDrawable(null)  // == getWhiteDot().drawable
+
+    private var flashingAction: FlashingAction? = null
+    private var flashingValue = 1f
+
+    private var label: Label? = null
+    //endregion
+
+    operator fun get(index: Int) = values[index]
+    operator fun set(index: Int, value: Float) {
+        values[index] = value.coerceIn(minValue, maxValue)
+        invalidate()
+    }
+
+    /** Shortcut to first value */
+    var value: Float
+        get() = values[0]
+        set(value) { values[0] = value }
+
+    /** Set [style] value [flashingSegment][HealthBarStyle.flashingSegment] **before** turning this on */
+    var allowAnimations = false
+        set(value) {
+            updateAnimations(value)
+            field = value
+        }
+
+    /** Special case for the UnitGroup - one segment colored by value and 1f black border */
+    fun setupForUnitOrCity(current: Int) {
+        style.background = BaseScreen.skinStrings.getUiBackground("General/HealthBar", tintColor = Color.BLACK)
+        style.padVertical = 1f
+        style.padHorizontal = 1.2f
+        values[0] = current.toFloat()
+        style.colors[0] = when {
+            current <= 33 -> Color.RED
+            current <= 66 -> Color.ORANGE
+            else -> Color.GREEN
```

Comment:
For cities that have >100 health, this is just wrong - we should be looking at *percentage of current health out of maximum*